CONTRIBUTING.rst

```diff
@@ -1,36 +1,56 @@
-Contributing
-============
-
 Instructions for contributors
 -----------------------------
 
+In order to make a clone of the `GitHub <https://github.com/aio-libs/aiohttp>`_ repo: open the link and press the "Fork" button on the upper-right menu of the web page.
 
-In order to make a clone of the GitHub_ repo: open the link and press the
-"Fork" button on the upper-right menu of the web page.
-
-I hope everybody knows how to work with git and github nowadays :)
+If you'd like to learn more about Git and GitHub, `check out GitHub's helpful introduction
+<https://docs.github.com/en/get-started/using-git/about-git>`_.
 
 Workflow is pretty straightforward:
 
-  1. Clone the GitHub_ repo using the ``--recurse-submodules`` argument
+  0. Make sure you are reading the latest version of this document.
+     It can be found in the GitHub_ repo in the ``docs`` subdirectory.
+
+  1. Clone your forked GitHub_ repo with the ``--recurse-submodules`` flag as shown in the command below,
+     ensuring to replace the placeholder with your github username:
+
+      .. code-block:: shell
+
+         $ git clone \
+            https://github.com/<your_github_username>/aiohttp.git \
+            --recurse-submodules
+
 
   2. Setup your machine with the required development environment
 
   3. Make a change
 
   4. Make sure all tests passed
 
-  5. Add a file into the ``CHANGES`` folder, named after the ticket or PR number
+  5. Add a file into the ``CHANGES`` folder (see `Changelog update <CHANGES>`_ for how).
 
   6. Commit changes to your own aiohttp clone
 
   7. Make a pull request from the github page of your clone against the master branch
 
   8. Optionally make backport Pull Request(s) for landing a bug fix into released aiohttp versions.
 
-.. important::
+.. note::
+
+   The project uses *Squash-and-Merge* strategy for *GitHub Merge* button.
+
+   Basically it means that there is **no need to rebase** a Pull Request against
+   *master* branch. Just ``git merge`` *master* into your working copy (a fork) if
+   needed. The Pull Request is automatically squashed into the single commit
+   once the PR is accepted.
+
+.. note::
+
+   GitHub issue and pull request threads are automatically locked when there has
+   not been any recent activity for one year.  Please open a `new issue
```

Comment:
I don't think this is true anymore (I see locked issues from years ago, but don't think there's any auto-locking in the past 4 years). Maybe just reword to open a new issue instead of commenting on a closed issue.

Reply:
Or we should re-establish the automation. It used to be run by https://github.com/apps/stale, but the app has been deprecated in favor of the action: https://github.com/marketplace/actions/close-stale-issues. We just never remembered to migrate.

Reply:
Personally, I find auto-locked issues a little annoying. Though if it's after a year then I don't think it makes a difference, we don't tend to see anything on issues that have been closed that long. For this PR, I think it just needs to document the current status.